src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -441,6 +442,9 @@ public unsafe bool TryReadTo(out ReadOnlySequence<T> sequence, ReadOnlySpan<T> d
                 ReadOnlySpan<T> next = Peek(peekBuffer);
                 if (next.SequenceEqual(delimiter))
                 {
+                    //TODO: Figure out a faster way to do this, potentially by avoiding the Advance in the previous TryReadTo call
```

Comment:
Fyi @JeremyKuhne, noticed a bug. I don't think this is the cleanest solution and there are likely other edge cases.